tiles/push_block.gd

```diff
@@ -0,0 +1,40 @@
+extends KinematicBody2D
+
+class_name PushBlock,"res://tiles/push_block.png"
+
+onready var tween := $Tween
+onready var ray := $RayCast2D
+
+var push_acumulator = 0
+
+func _process(delta):
+	if push_acumulator>0:
+		push_acumulator -= delta
+	if push_acumulator < 0:
+		push_acumulator == 0
+
+
+func push(delta,from:Vector2):
+	var relative_pos = (global_position - from).snapped(Vector2(16,16))
+	ray.cast_to = relative_pos.normalized()*16
+	push_acumulator = push_acumulator+2*delta
+	if push_acumulator<1: return
+	if tween.is_active(): return
+	if ray.get_collider(): return
+	# We need to take only player position and not how he move otherwize we can trick the block to move up!
+	
+	if abs(relative_pos.x) > abs(relative_pos.y):
```

Comment:
Similarly here, we can extract this statement into a small function named in a way that describes the comment above. Allows us to remove the comment. This class isn't very big so adding a couple small readability functions isn't a big deal.

Reply:
No, the comment is not explaining **what**, but it is explaining **why**. No functions can replace that kind of comments.

This comment came into play because I first implemented it using the "direction" of the player then saw it could be abused. You could move diagonally while rubbing a side of the block and the block would start to move perpendicularly to the player. So, I had to chose another way and this is why I take only player relative position so I can know on which side he is pushing and deduce where the block should go.

Reply:
```python
func push(delta : float, pushing_force : Vector2) -> void:
	var relative_direction = (global_position - pushing_force).snapped(Vector2(16,16))
	ray.cast_to = relative_direction.normalized()*16
	push_acumulator = push_acumulator+2*delta
	if meets_bailout_criteria(): return

	tween_move(convert_to_push_directin(relative_direction))

func meets_bailout_criteria() -> bool:
	if push_acumulator<1: return true
	if tween.is_active(): return true
	if ray.get_collider(): return true

	return false

func convert_to_push_directin(relative_direction : Vector2) -> Vector2:
	if is_being_pushed_from_side(relative_direction):
		if relative_direction.x>0:
			return Vector2.RIGHT*16
		else:
			return Vector2.LEFT*16
	else:
		if relative_direction.y>0:
			return relative_direction.DOWN*16
		else:
			return relative_direction.UP*16
			
	return Vector2.Zero

func is_being_pushed_from_side(relative_direction : Vector2) -> bool:
	# We need to take only player position and not how he move otherwize we can trick the block to move up!
	return abs(relative_direction.x) > abs(relative_direction.y)

Reply:
Comment remains to give insight to line but functions are extracted to make the main push method read much faster, the function names also give you a heads up for what you expect to see. Big fan of extracting logic into smaller functions

Reply:
**meets_bailout_criteria** name screams "I had to make a function out of that". Why are we bailing out ?

The 3 if could also be condensed to:

`if push_acumulator<1 || tween.is_active() || ray.get_collider(): return`

Why did I chose to create 3 "if" instead ?

**is_being_pushed_from_side** is not doing that, it checks if player is closer to the X or Y object axis. Object is always **being pushed from sides**.

This PR can be merged and the changes done as you see fit. Though I only see increased mental burden and getting away of idioms that are there to relieve the developer mind.